syncstorage/views/__init__.py

```diff
@@ -319,6 +342,84 @@ def post_collection(request):
     return res
 
 
+def post_collection_transaction(request):
+    storage = request.validated["storage"]
+    userid = request.validated["userid"]
+    collection = request.validated["collection"]
+    bsos = request.validated["bsos"]
+    invalid_bsos = request.validated["invalid_bsos"]
+    batch = request.validated["batch"]
+    commit = request.validated["commit"]
+
+    # The "batch" key is set only on a multi-POST batch request prior to a
+    # commit.  The "modified" key is only set upon a successful commit.
+    # The two flags are mutually exclusive.
+    # Any failures at all mean cancelling the transaction completely.
+    res = {'success': [], 'failed': {}}
+
+    # If there are any parsing failures, we won't even start a transaction.
+    if len(invalid_bsos):
+        for (id, error) in invalid_bsos.iteritems():
+            res["failed"][id] = error
+        # if batch and batch is not True:
+        #     storage.delete_transaction(batch)
+        return res
+
+    try:
+        if batch is True:
+            try:
+                batch = storage.create_transaction(userid, collection)
+            except ConflictError, e:
+                # ConflictError here means a client is spamming requests,
+                # I think.
+                logger.error('Collision in transaction creation!')
+                logger.error(e)
+                raise HTTPServiceUnavailable('Too many transactions opened')
```

Comment:
I think we should raise the original ConflictError here, and it'll be converted into an appropriate response by the middleware layer (I believe it's a 503)